### PR TITLE
Fixed issue when unauthorized the registered app

### PR DESCRIPTION
**Pull request by Kanapriya**
## Purpose
Resolve the issue when unauthorized the user by clicking on "Decline" button in registered app.

## Documentation
Documentation added with this PR inside /docs diectory

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
JDK 1.8
Ubuntu
Mozilla
 
## Learning

### CODE REVIEW

File: component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticatorConstants.java [hunk 1]
```diff
@@ -64,4 +64,7 @@
     public static final String REDDIT_CLIENT_SECRET = "Client Secret";
     //Reddit value for Callback_url
     public static final String REDDIT_CALLBACK_URL = "Callback URL";
+    public static final String OAUTH2_PARAM_ERROR = "error";
+    public static final String ERROR = "error: ";
+    public static final String STATE = ", state: ";
 }
```
Comment (Nirthika) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticatorConstants.java]: Please add a new line here

File: component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java [hunk 2]
```diff
@@ -128,7 +129,29 @@ public String getScope(String scope, Map<String, String> authenticatorProperties
      * @return weather true or false
      */
     public boolean canHandle(HttpServletRequest httpServletRequest) {
-        return httpServletRequest.getParameter(RedditAuthenticatorConstants.CODE) != null;
+        return httpServletRequest.getParameter(RedditAuthenticatorConstants.CODE) != null || httpServletRequest
+                .getParameter(RedditAuthenticatorConstants.OAUTH2_PARAM_ERROR)!=null;
```
Comment (Megala21) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java]: Formatting issue !=

File: component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java [hunk 3]
```diff
@@ -128,7 +129,29 @@ public String getScope(String scope, Map<String, String> authenticatorProperties
...
+     */
+    private void handleErrorResponse(HttpServletRequest request) throws InvalidCredentialsException {
+        if (request.getParameter(RedditAuthenticatorConstants.OAUTH2_PARAM_ERROR) != null) {
+            StringBuilder errorMessage = new StringBuilder();
+            String error = request.getParameter(RedditAuthenticatorConstants.OAUTH2_PARAM_ERROR);
+            String state = request.getParameter(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE);
+            errorMessage.append(RedditAuthenticatorConstants.ERROR).append(error)
+                    .append(RedditAuthenticatorConstants.STATE).append(state);
+            if (log.isDebugEnabled()) {
+                log.debug("Failed to authenticate via reddit when unauthorized the registered app. " +
```
Comment (hasuniea) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java]: Here debug log should be meaning full. Can you change it as 
log.debug("Failed to authenticate via Reddit when unauthorized the registered app : " +  errorMessage.toString())

File: component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java (hunk 2 above)
Comment (hasuniea) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java]: Format the code.

File: component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticatorConstants.java (hunk 1 above)
Comment (Kanapriya) [reply component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticatorConstants.java]: Fixed in 712cd7b17f204023cc7106222b39e945853f010f

File: component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java (hunk 2 above)
Comment (Kanapriya) [reply component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java]: Fixed in 712cd7b17f204023cc7106222b39e945853f010f

File: component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java (hunk 3 above)
Comment (Kanapriya) [reply component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java]: Fixed in 712cd7b17f204023cc7106222b39e945853f010f

File: component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java (hunk 2 above)
Comment (Kanapriya) [reply component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java]: Fixed in 712cd7b17f204023cc7106222b39e945853f010f

**Review by hasuniea — APPROVED** [review]

**Review by Nirthika — APPROVED** [review]

**Review by Megala21 — APPROVED** [review]

**Pull request merged**